Inference custom model example (#315)
mosaicml/examples: examples/inference/README.md

# Inference with Custom Models on the MosaicML platform

[MosaicML Inference]() enables easy deployment of machine learning models. In this folder, we provide an example for how to create and deploy your own model on the MosaicML platform.

You'll find in this folder:

> - `hf_instructor_model_class.py` - a custom model class that implements the Hugging Face large instructor

Reviewer: ```suggestion
- `hf_instructor_model_class.py` - a custom model class that implements the Hugging Face model [instructor-large](https://huggingface.co/hkunlp/instructor-large)
```

- `custom_mosaic_gpt.yaml` - a script to deploy the model on the MosaicML platform

## Prerequisites

> The only thing you need to deploy this model is access to MosaicML's inference service. [Contact us to get started]().

Reviewer: ```suggestion
The only thing you need to deploy this model is access to MosaicML's inference service. [Contact us to get started](https://forms.mosaicml.com/demo).
```
(Note that this URL might change in like a day)

## Configuring the Model

> Currently, we offer support for any hugging face model and custom models that follow a model class spec. The provided `.yaml` file is configured to deploy the custom instructor model class. Information on the configuration options can be found [here]().

Reviewer: ```suggestion
Currently, we offer support for any Hugging Face model and custom models that implement a particular interface. The provided `.yaml` file is configured to deploy the custom instructor model class. Information on the configuration options can be found [here]().
```
Can we document this interface here? Or is there a class they have to subclass or something? Mostly just want to link to something specific enough for people to use it.

Author: Its a class they have to inherit from. I've added a link to the mcli inference docs where the class is listed

# Deploying the Model

> Deploying the model is as simple as running `mcli deploy -f yamls/instructor_model.yanl`.

Reviewer: ```suggestion
Deploying the model is as simple as running `mcli deploy -f yamls/instructor_model.yaml`.
```
